include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
+#define GSL_USE_STD_BYTE 1
+
+#else
+
+#define GSL_USE_STD_BYTE 0
+
+#endif // _HAS_STD_BYTE
+
 #endif           // _MSC_VER
```

Comment:
(...and leave an obvious place for people to tack on support for other standard libraries if they so desire.)